optuna/visualization/matplotlib/_edf.py

```diff
@@ -1,9 +1,144 @@
+from optuna.trial import TrialState
 from optuna.visualization.matplotlib._matplotlib_imports import _imports
 
 
 if _imports.is_successful():
     from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
+
+_logger = get_logger(__name__)
+
+
+def plot_edf(study: Union[Study, Sequence[Study]]) -> Axes:
+    """Plot the objective value EDF (empirical distribution function) of a study with Matplotlib.
+
+    Note that only the complete trials are considered when plotting the EDF.
+
+    .. note::
+
+        EDF is useful to analyze and improve search spaces.
+        For instance, you can see a practical use case of EDF in the paper
+        `Designing Network Design Spaces <https://arxiv.org/abs/2003.13678>`_.
+
+    .. note::
+
+        The plotted EDF assumes that the value of the objective function is in
+        accordance with the uniform distribution over the objective space.
+
+    Example:
+
+        The following code snippet shows how to plot EDF.
+
+        .. testcode::
+
+            import math
+
+            import optuna
+
+
+            def ackley(x, y):
+                a = 20 * math.exp(-0.2 * math.sqrt(0.5 * (x ** 2 + y ** 2)))
+                b = math.exp(0.5 * (math.cos(2 * math.pi * x) + math.cos(2 * math.pi * y)))
+                return -a - b + math.e + 20
+
+
+            def objective(trial, low, high):
+                x = trial.suggest_float("x", low, high)
+                y = trial.suggest_float("y", low, high)
+                return ackley(x, y)
+
+
+            sampler = optuna.samplers.RandomSampler()
+
+            # Widest search space.
+            study0 = optuna.create_study(study_name="x=[0,5), y=[0,5)", sampler=sampler)
+            study0.optimize(lambda t: objective(t, 0, 5), n_trials=500)
+
+            # Narrower search space.
+            study1 = optuna.create_study(study_name="x=[0,4), y=[0,4)", sampler=sampler)
+            study1.optimize(lambda t: objective(t, 0, 4), n_trials=500)
+
+            # Narrowest search space but it doesn't include the global optimum point.
+            study2 = optuna.create_study(study_name="x=[1,3), y=[1,3)", sampler=sampler)
+            study2.optimize(lambda t: objective(t, 1, 3), n_trials=500)
+
+            optuna.visualization.plot_edf([study0, study1, study2])
+
+    Args:
+        study:
+            A target :class:`~optuna.study.Study` object.
+            You can pass multiple studies if you want to compare those EDFs.
+
+    Returns:
+        A :class:`matplotlib.figure.Figure` object.
+    """
+
+    _imports.check()
+
+    if isinstance(study, Study):
+        studies = [study]
+    else:
+        studies = list(study)
+
+    return _get_edf_plot(studies)
+
+
+def _get_edf_plot(studies: List[Study]) -> Axes:
+
+    # Set up the graph style.
+    plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
+    fig, ax = plt.subplots()
```

Comment:
It seems that we don't need `fig`.
```suggestion
    ax = plt.subplot()
```

Reply:
How about this?
`plt.subplot` (not `plt.subplots`) is for adding a subplot to an existing plot.

```suggestion
    _, ax = plt.subplots()
```

Reply:
Thanks for the confirmation. LGTM.